fairlearn/reductions/_moments/error_rate.py

```diff
@@ -9,14 +9,58 @@
 from fairlearn._input_validation import _validate_and_reformat_input
 
 
+_MESSAGE_BAD_COSTS = "costs needs to be a dictionary with keys " \
+    "'fp' and 'fn' containing non-negative values, which are not both zero"
+
+
 class ErrorRate(ClassificationMoment):
-    """Misclassification error."""
+    r"""Misclassification error as a moment.
+
+    A classifier :math:`h(X)` has the misclassification error equal to
+
+    .. math::
+      P[h(X) \ne Y]
+
+    It is also possible to specify costs for false positives and false
+    negatives. The error then evaluates to
+
+    .. math::
+      c_{FP} P[h(X)=1, Y=0] + c_{FN} P[h(X)=0, Y=1]
+
+    where :math:`c_{FP}` and `c_{FN}` are the costs of false positive
+    and false negative errors respectively. The standard misclassification
+    error corresponds to :math:`c_{FP}=c_{FN}=1.0`.
+
+    Parameters
+    ----------
+    costs : dict
+        The dictionary with keys ``"fp"`` and ``"fn"`` containing the
+        costs of false positives and false negatives. If none are provided
+        costs of 1.0 are assumed.
+    """
 
     short_name = "Err"
 
+    def __init__(self, *, costs=None):
+        """Initialize the costs."""
+        super(ErrorRate, self).__init__()
+        if costs is None:
+            self.fp_cost = 1.0
+            self.fn_cost = 1.0
+        elif (type(costs) is dict
+              and costs.keys() == {'fp', 'fn'}
```

Comment:
The strings `fp` and `fn` should be made into constants.

Reply:
Is this a deal-breaker? We had this conversation elsewhere and made (more or less?) the choice that literal string = constants, e.g., https://github.com/fairlearn/fairlearn/blob/main/fairlearn/postprocessing/_threshold_optimizer.py.

Reply:
I hadn't noticed those slipping in to `_threshold_optimizer.py`. Strings which pop up more than once really ought to be constants.

Reply:
Is this required for this PR or should we hold off to a separate conversation re. coding style?

I'm really hesitant to create a constant here. Replacing `'fp'` and `'fn'` by `FP` and `FN` or something of the sort is going to make the code harder to read in my opinion. Right now, the constructor, its documentation and validation are very near-each other in the code.

Besides `ThresholdOptimizer`, we similarly don't create constants for the method argument in `MetricFrame.difference` and there are probably other places in the library.

Reply:
OK. The point about us not doing constants like that elsewhere is fair.